files/audit.sh

```diff
@@ -0,0 +1,178 @@
+for ignore_node in "${NODES_TO_IGNORE[@]}"; do
+    for i in "${!NODES_IN_ALL_GRP[@]}"; do
+        if [[ ${NODES_IN_ALL_GRP[$i]} == "$ignore_node" ]]; then
+            unset 'NODES_IN_ALL_GRP[i]'
+        fi
+    done
+done
+NODES=("${NODES_IN_ALL_GRP[@]}")
+
+if [[ $verbose ]]; then
+    if lsdef  -t site -i excludenodes | grep -q excludenodes; then
+        echo "Nodes ignored by being in excludenodes site table:"
+        lsdef  -t site -i excludenodes | grep excludenodes
+    fi
+
+    if [[ ${#NODES_TO_IGNORE[@]} -ne 0 ]]; then
+        echo "Nodes ignored by being set in NODES_TO_IGNORE:"
+        for i in "${NODES_TO_IGNORE[@]}"; do printf "%s " "$i"; done
+        echo
+    fi
+fi
+
+# List of nodes comma delimited
+NODES_CSV=$(IFS=, ; echo "${NODES[*]}")
+# Run audit script
+/opt/xcat/bin/xdsh "${NODES_CSV}" -t 30 "$AUDIT_SCRIPT $AUDIT_USER $AUDIT_ID"
+
+# Run module audit if requested
+if [[ $module_audit ]]; then
+    ssh "$MODULE_HOST" "find ${MODULE_DIRS} -type f" > "$REPORT_DST/audit-MODULES"
+fi
+
+# Not as fast as xdcp, but gets us the filenames we need
+for n in "${NODES[@]}";
+do
+   scp "$n:/tmp/audit-${n}-${AUDIT_ID}" "$REPORT_DST/audit-${n}"
+done
+
+if [[ $automated ]]; then
+    # Update symlink pointers to keep track of previous audit and current audit
+    if [[ -h $AUTOMATED_REPORT_DST/current ]]; then
+        [[ -h $AUTOMATED_REPORT_DST/previous ]] && unlink "$AUTOMATED_REPORT_DST/previous"
+        ln -s "$(readlink -f $AUTOMATED_REPORT_DST/current)" "$AUTOMATED_REPORT_DST/previous"
+        unlink "$AUTOMATED_REPORT_DST"/current
+    fi
+    ln -s "$REPORT_DST" "$AUTOMATED_REPORT_DST/current"
+
+    if [[ $email_compare ]]; then
+        prev=$(readlink -f "$AUTOMATED_REPORT_DST"/previous)
+        curr=$(readlink -f "$AUTOMATED_REPORT_DST"/current)
+
+        # Test if dirs exist
+        [[ ! -d ${prev} || ! -d ${curr} ]] && email_and_exit "ERROR" "ERROR: One or both of these dirs did not exist when trying to compare:" "${prev}" "${curr}"
+
+        # Run compare and save output
+        COMPARE_OUTPUT=$(${COMPARE_SCRIPT} "${prev}" "${curr}") || email_and_exit "ERROR" "ERROR (exit code $?) while running:" "${COMPARE_SCRIPT} ${prev} ${curr}"
+
+        # Test if there is no difference found
+        [[ -z $COMPARE_OUTPUT ]] && email_and_exit "No Diff" "No differences detected between these audits:" "${prev}" "${curr}"
+
+        # If we make it this far the audit ran ok and there was a difference in the audits
+        email_and_exit "Diff Detected" "There was a difference detected between these audits:" "${prev}" "${curr}" "" "Difference is:" "" "${COMPARE_OUTPUT}"
+    fi
+
+else
+    # Run manually, so tar up results and stash in /tmp for an admin to copy somewhere else
+    tar -czvf "/tmp/${DATETIME}-${AUDIT_USER}_RH8.tgz" "$REPORT_DST"
+    chown "$AUDIT_USER" "/tmp/${DATETIME}-${AUDIT_USER}_RH8.tgz"
+    echo "A tar'd up report was saved to : /tmp/${DATETIME}-${AUDIT_USER}_RH8.tgz"
```

Comment:
What's the reasoning for hard-coding _RH8 here? If we need that identifier can we programmatically pull it?

Reply:
Good catch. The identifier was needed on places like mforge where we need to run two audits since one admin node can't reach both RH7 and RH8 nodes.

I've removed the hardcoding and instead make the identifier the hostname of the xcat server.